test/unit/gateways/cyber_source_test.rb

```diff
@@ -121,6 +121,41 @@ def test_merchant_description
     end.respond_with(successful_purchase_response)
   end
 
+  def test_uses_names_from_billing_address_if_present
+    name = 'Wesley Crusher'
+
+    stub_comms do
+      @gateway.authorize(100, @credit_card, billing_address: { name: name })
+    end.check_request do |_endpoint, data, _headers|
+      assert_match(%r(<billTo>.*<firstName>Wesley</firstName>.*</billTo>)m, data)
+      assert_match(%r(<billTo>.*<lastName>Crusher</lastName>.*</billTo>)m, data)
+    end.respond_with(successful_purchase_response)
+  end
+
+  def test_uses_names_from_shipping_address_if_present
+    name = 'Wesley Crusher'
+
+    stub_comms do
+      @gateway.authorize(100, @credit_card, shipping_address: { name: name })
+    end.check_request do |_endpoint, data, _headers|
+      assert_match(%r(<shipTo>.*<firstName>Wesley</firstName>.*</shipTo>)m, data)
+      assert_match(%r(<shipTo>.*<lastName>Crusher</lastName>.*</shipTo>)m, data)
+    end.respond_with(successful_purchase_response)
+  end
```

Comment:
Do we need to test that, when both billing and shipping address are present, each is sourced from the proper address? Eg:
```suggestion
  def test_uses_names_from_billing_and_shipping_addresses_when_present
    billing_name = 'Wesley Crusher'
    shipping_name = 'Elvis Preisler'

    stub_comms do
      @gateway.authorize(100, @credit_card, billing_address: { name: billing_name }, shipping_address: { name: shipping_name })
    end.check_request do |_endpoint, data, _headers|
      assert_match(%r(<billTo>.*<firstName>Wesley</firstName>.*</billTo>)m, data)
      assert_match(%r(<billTo>.*<lastName>Crusher</lastName>.*</billTo>)m, data)
      assert_match(%r(<shipTo>.*<firstName>Elvis</firstName>.*</shipTo>)m, data)
      assert_match(%r(<shipTo>.*<lastName>Preisler</lastName>.*</shipTo>)m, data)
    end.respond_with(successful_purchase_response)
  end
```